coffee_github/src/utils.rs

```diff
@@ -24,46 +25,25 @@ pub async fn clone_recursive_fix(repo: git2::Repository, url: &URL) -> Result<()
     Ok(())
 }
 
-pub fn fast_forward(path: &str, branch: &str) -> Result<UpgradeStatus, CoffeeError> {
-    let repo = git2::Repository::open(path).map_err(|err| error!("{}", err.message()))?;
-
-    repo.find_remote("origin")
-        .map_err(|err| error!("{}", err.message()))?
-        .fetch(&[branch], None, None)
-        .map_err(|err| error!("{}", err.message()))?;
-
-    let fetch_head = repo
-        .find_reference("FETCH_HEAD")
-        .map_err(|err| error!("{}", err.message()))?;
+pub async fn git_upgrade(
+    path: &str,
+    branch: &str,
+    verbose: bool,
+) -> Result<UpgradeStatus, CoffeeError> {
+    use tokio::process::Command;
 
-    let fetch_commit = repo
-        .reference_to_annotated_commit(&fetch_head)
-        .map_err(|err| error!("{}", err.message()))?;
-
-    let analysis = repo
-        .merge_analysis(&[&fetch_commit])
-        .map_err(|err| error!("{}", err.message()))?;
+    let repo = git2::Repository::open(path).map_err(|err| error!("{}", err.message()))?;
 
-    if analysis.0.is_up_to_date() {
-        Ok(UpgradeStatus::UpToDate)
-    } else if analysis.0.is_fast_forward() {
-        let refname = format!("refs/heads/{}", branch);
-        let mut reference = repo
-            .find_reference(&refname)
-            .map_err(|err| error!("{}", err.message()))?;
+    let (local_commit, _) = get_repo_info!(repo);
 
-        reference
-            .set_target(fetch_commit.id(), "Fast-Forward")
-            .map_err(|err| error!("{}", err.message()))?;
+    let cmd = format!("git pull origin {branch}");
```

Comment:
I have a concern here 
Maybe some developer force pushed a commit and now `git pull origin {branch}` would return an error

we can do 
```bash
git fetch origin
git reset --hard origin/{branch}
``` 
to account for this case

Reply:
we can do a `git pull origin {branch} --force` but Idk if your is another way to do the same things?

Reply:
Good suggestion, thanks!